tools/umi_tools/umi_tools.yml

```diff
@@ -0,0 +1,45 @@
+name: umi_tools
+version: 1.0
+description: Tools for dealing with Unique Molecular Identifiers (UMIs)/Random Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes. 
+keywords:
+    - UMI
+    - RMT
+    - Barcode
+tools:
+    - umi_tools:
+        description: |
+            Tools for dealing with Unique Molecular Identifiers (UMIs)/Random Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes. 
+        homepage: https://github.com/CGATOxford/UMI-tools
+        documentation: https://umi-tools.readthedocs.io/en/latest/
+processes:
+    - dedup:
+        operation: |
+            Set command args to params.umitools_dedup_args
+            The program will execute with the following pattern:
+                umi_tools dedup --log={SAMPLE_ID}.dedup.log {params.umitools_dedup_args} -I {SAMPLE_ID}.bam -S {SAMPLE_ID}.dedup.bam --output-stats={SAMPLE_ID}
+        description: |
+            Groups PCR duplicates and de-duplicates reads to yield one read per group. 
+            Use this when you want to remove the PCR duplicates prior to any downstream analysis.
+        input:
```

Comment:
I feel a bit nervous about this part of the yaml file. I'm not really sure how it will be used yet and the syntax is very difficult. Notably, this should be a nested list with two `-` so that there is one list per input which can in turn have multiple parts. See for example:

https://github.com/nf-core/modules/blob/9dee710b043458c62726d025c903bf185fec4ad1/tools/cutadapt/meta.yml#L20-L26

I wonder if we should strip this all from the yaml file and come back to it later with a JSON Schema approach instead.

Thoughts?

Reply:
I also think it could cause problems later on - this module will have 6 processes in it by the end, it could look very messy.

Reply:
You mean the YAML could cause problems? Would JSON Schema be ok? It's a lot more verbose, would probably need some new tooling to help developers write it.

I wonder if we just stick with documentation for now instead of scripted descriptions.

Reply:
whats the aim with the file? is it to auto-generate documentation at a later date?

Reply:
Yes basically. I wonder if we can do the same with just code comments or something instead though.. 🤔